src/Extensions/Hosting.Services.Web/KestrelListenerBuilder.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Fabric;
+using System.IO;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.ServiceFabric.Services.Communication.AspNetCore;
+using Microsoft.ServiceFabric.Services.Communication.Runtime;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web
+{
+	/// <summary>
+	/// Creates ServiceInstanceListener with all of Omex dependencies initialized
+	/// </summary>
+	internal sealed class KestrelListenerBuilder<TStartup, TServiceContext> : IListenerBuilder<TServiceContext>
+		where TStartup : class
+		where TServiceContext : ServiceContext
+	{
+		public string Name { get; }
+
+
+		public ICommunicationListener Build(TServiceContext context) =>
+			new KestrelCommunicationListener(context, Name, (url, listener) => BuildWebHost(context, url, listener));
+
+
+		internal KestrelListenerBuilder(
+			string name,
+			ServiceFabricIntegrationOptions options,
+			Action<IWebHostBuilder> builderExtension)
+		{
+			Name = name;
```

Comment:
Validate this for empty or whitespace

Reply:
Empty or whitespace is allowed values for Sf listener name and I would like to avoid complicating code with unnecessary validation, sorry